src/core/server/elasticsearch/client/mocks.ts

```diff
@@ -163,7 +163,7 @@ function createApiResponse<TResponse = Record<string, any>>(
   return {
     body: {} as any,
     statusCode: 200,
-    headers: {},
+    headers: { 'x-elastic-product': 'Elasticsearch' },
```

Comment:
Adds the header as a default (as per https://github.com/elastic/kibana/pull/107301#discussion_r681922905) that can be overridden if needed in unit tests.